core/server/src/main/java/alluxio/master/file/FileSystemMasterClientServiceHandler.java

```diff
@@ -125,11 +127,20 @@ public Void call() throws AlluxioException, IOException {
   }
 
   @Override
-  public void free(final String path, final boolean recursive) throws AlluxioTException {
+  public void free(final String path, final boolean recursive, final FreeTOptions options)
+      throws AlluxioTException {
     RpcUtils.call(new RpcCallable<Void>() {
       @Override
       public Void call() throws AlluxioException {
-        mFileSystemMaster.free(new AlluxioURI(path), recursive);
+        if (options == null) {
+          // For Alluxio client v1.4 or earlier.
+          // NOTE, we try to be conservative here so early Alluxio clients will not be able to force
+          // freeing pinned items but see the error thrown.
```

Comment:
I and @jsimsa  had a discussion on this and Jiri convinced me freeing pinned files silently for older client without telling users is bad. So the old client will experience different free result on pinned files when it talks to new server (which is anyway a corner case IMO). Let me know if you think differently.